src/odemis/driver/npmc.py

```diff
@@ -0,0 +1,845 @@
+        logging.info("Found Newport ESP301 device on port %s, Ver: %s",
+                     self._port, self._version)
+
+        self._offset = {}
+        self._axis_conv_factor = {}
+        self._axis_unit = {}
+
+        # Not to be mistaken with axes which is a simple public view
+        self._axis_map = {}  # axis name -> axis number used by controller
+        axes_def = {}  # axis name -> Axis object
+        position = {}
+        speed = {}
+        accel = {}
+        decel = {}
+        self._id = {}
+
+        for axis_name, axis_par in axes.items():
+            # Unpack axis parameters from the definitions in the YAML
+            try:
+                axis_num = axis_par['number']
+            except KeyError:
+                raise ValueError("Axis %s must have a number to identify it. " % (axis_name,))
+
+            try:
+                axis_range = axis_par['range']
+            except KeyError:
+                logging.info("Axis %s has no range. Assuming (-1, 1)", axis_name)
+                axis_range = (-1, 1)
+
+            try:
+                axis_unit = axis_par['unit']
+            except KeyError:
+                logging.info("Axis %s has no unit. Assuming m", axis_name)
+                axis_unit = "m"
+
+            if axis_unit not in ('m', 's'):
+                raise ValueError("Invalid axis unit. Should be m or s.")
+
+            try:
+                conv_factor = float(axis_par['conv_factor'])
+            except KeyError:
+                logging.info("Axis %s has no conversion factor. Assuming 1000 (m to mm)", axis_name)
+                conv_factor = 1000.0
+
+            self._axis_map[axis_name] = axis_num
+            self._axis_conv_factor[axis_num] = conv_factor
+            self._axis_unit[axis_num] = axis_unit
+            self._id[axis_num] = self.GetIdentification(axis_num)
+            speed[axis_name] = self.GetSpeed(axis_num)
+            accel[axis_name] = self.GetAcceleration(axis_num)
+            decel[axis_name] = self.GetDeceleration(axis_num)
+
+            # Force millimetres for consistency as the internal unit.
+            self.SetAxisUnit(axis_num, "mm")
+            # initialize each motor
+            self.MotorOn(axis_num)
+
+            ad = model.Axis(canAbs=False, unit=axis_unit, range=axis_range)
+            axes_def[axis_name] = ad
+
+        model.Actuator.__init__(self, name, role, axes=axes_def, **kwargs)
+
+        self._hwVersion = str(self._id)
+        self._swversion = self._version
+
+        # Get the position in object coord with the offset applied.
+
+        # RO, as to modify it the client must use .moveRel() or .moveAbs()
+        self.position = model.VigilantAttribute({}, readonly=True)
+        self._updatePosition()
+
+        self._speed = speed
+        self._accel = accel
+        self._decel = decel
+
+        # set offset due to mounting of components (float)
+        self._metadata[model.MD_POS_COR] = None
```

Comment:
Please initialise this value with a valid value (eg, `{}`). In practice, it's not necessary, but it helps understanding what should be the type.